File: frontend/src/services/uploadService.js
import api from './api'

const unwrap = (response) => response?.data?.data ?? response?.data ?? response

export const uploadService = {
  async uploadImage(file, folder, entityId) {
    const formData = new FormData()
    formData.append('file', file)
    formData.append('folder', folder)

    if (entityId !== undefined && entityId !== null) {
      formData.append('entity_id', String(entityId))
    }

    const response = await api.post('/uploads/images', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
    return unwrap(response)
  },

  async deleteImage(publicId) {
    const response = await api.delete('/uploads/images', {
      data: { public_id: publicId },
    })
    return unwrap(response)
  },
}

File: src/main/java/com/autowashpro/service/UploadService.java
package com.autowashpro.service;

import com.autowashpro.dto.response.UploadImageResponse;
import com.autowashpro.entity.Upload;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface UploadService {
    UploadImageResponse uploadImage(
            MultipartFile file,
            String folder,
            Long entityId,
            Long currentUserId,
            String role);

    void deleteImage(String publicId, Long currentUserId, String role);

    List<Upload> requireInspectionUploads(
            Long bookingId,
            List<String> publicIds,
            Long currentInspectionId);
}
